### PR TITLE
Added "Ignore Update Errors" and number attribute options

### PR DESCRIPTION
**Pull request by jodyfarnden**
I needed to be able to migrate data in entities that had values in read-only records.
To do this I have implemented an "Ignore Update Errors" catch that continues with the requests even if one is unable to complete successfully. This covers bad data (too long for new field length or wrong format) and data that may be inserted into read only records causing an error. Because this may result in data loss i have added a warning message box that appears when the checkbox is ticked.

I also needed to be able to specify the precision on a newly created decimal field, so have adjusted the UI to perform the hide/show actions for number attributes when the number types (Decimal, whole number, float, currency) are selected in the new attribute area. The values are retrieved when the new attribute settings are created and applied to the new attribute creation process.

### CODE REVIEW

File: DLaB.AttributeManager/Logic.cs [hunk 1]
```diff
@@ -68,6 +69,12 @@ public Logic(IOrganizationService service, ConnectionDetail connectionDetail, En
             Metadata = metadata;
         }
 
+        public Logic(IOrganizationService service, ConnectionDetail connectionDetail, EntityMetadata metadata, string tempPostFix, bool migrateData, bool ignoreUpdateErrors):
```
Comment (daryllabar) [inline DLaB.AttributeManager/Logic.cs]: Why create a separate Constructor?

File: DLaB.AttributeManager/DLaB.AttributeManager.csproj [hunk 2]
```diff
@@ -253,8 +253,8 @@ if $(ConfigurationName) == Release "$(SolutionDir)References\SetAssemblyVersion.
 </PreBuildEvent>
   </PropertyGroup>
   <PropertyGroup>
-    <PostBuildEvent>DEL /Q "$(TargetDir)MscrmTools\XrmToolBox\Plugins\*.*"
-xcopy "$(TargetDir)DLaB.*.dll" "$(TargetDir)MscrmTools\XrmToolBox\Plugins\" /I /Y
+    <PostBuildEvent>DEL /Q "$(TargetDir)XrmToolBox\Plugins\*.*"
```
Comment (daryllabar) [inline DLaB.AttributeManager/DLaB.AttributeManager.csproj]: Why was this changed?

File: DLaB.AttributeManager/Logic.cs [hunk 3]
```diff
@@ -1134,14 +1147,33 @@ private void PerformUpdates(IOrganizationService service, OrganizationRequestCol
...
+                        service.Update(((UpdateRequest)request).Target);
+                    } catch (Exception e)
+                    {
+                        if (IgnoreUpdateErrors)
+                        {
+                            countFaults = countFaults + 1;
+                        }
+                        else
+                        {
+                            throw new Exception(e.Message);
```
Comment (daryllabar) [inline DLaB.AttributeManager/Logic.cs]: Why wouldn't this just a be a ```throw e```?

File: DLaB.AttributeManager/AttributeManagerPlugin.cs [hunk 4]
```diff
@@ -1043,6 +1146,35 @@ private void SelectAllKeyDownHandler(object sender, KeyEventArgs e)
                 e.Handled = true;
             }
         }
+
+        private void grpSteps_Enter(object sender, EventArgs e)
```
Comment (daryllabar) [inline DLaB.AttributeManager/AttributeManagerPlugin.cs]: Can you remove this?

File: DLaB.AttributeManager/AttributeManagerPlugin.cs [hunk 5]
```diff
@@ -1043,6 +1146,35 @@ private void SelectAllKeyDownHandler(object sender, KeyEventArgs e)
...
+            }
+            return number;
+        }
+
+        private void chkIgnoreUpdateErrors_CheckedChanged(object sender, EventArgs e)
+        {
+            if (chkIgnoreUpdateErrors.Checked)
+            {
+                //warn about possible data loss
+                DialogResult result = MessageBox.Show("Selecting this option may result in significant data loss if there are read-only records or invalid data", "Warning", MessageBoxButtons.OKCancel, MessageBoxIcon.Warning);
```
Comment (daryllabar) [inline DLaB.AttributeManager/AttributeManagerPlugin.cs]: This just means that some of the data will not get migrated, because of the error, and just continue correct?

File: DLaB.AttributeManager/AttributeManagerPlugin.cs (hunk 4 above)
Comment (jodyfarnden) [reply DLaB.AttributeManager/AttributeManagerPlugin.cs]: This looks like an accidental creation while editing the UI. I have removed in latest changes.
